flake.nix

```diff
@@ -2,204 +2,267 @@
+                  #   glibcLocales # disabled to avoid error:  A definition for option `packages."[definition 4-entry 16]"' is not of type `package'.
+                  # ]
+                  ++ optional stdenv.isDarwin terminal-notifier
+                  ++ optionals stdenv.isDarwin (
+                    with darwin.apple_sdk.frameworks;
+                    [
+                      CoreFoundation
+                      CoreServices
+                    ]
+                  );
+                enterShell = ''
+                  export LOCALES="${cldr}/priv/cldr";
+                  export PORT="4000"
+                  export ENCRYPTION_KEY="your_secure_encryption_key_here"
+                  export DATABASE_USER="teslamate"
+                  export DATABASE_PASS="your_secure_password_here"
+                  export DATABASE_NAME="teslamate"
+                  export DATABASE_HOST="127.0.0.1"
+                  export DATABASE_PORT="${toString postgres_port}"
+                  export MQTT_HOST="127.0.0.1"
+                  export MQTT_PORT="${toString mosquitto_port}"
+                  export RELEASE_COOKIE="1234567890123456789"
+                  export TZDATA_DIR="$PWD/tzdata"
+                  export MIX_REBAR3="${rebar3}/bin/rebar3";
+                  mix deps.get
                 '';
-                postgres.enable = true;
-                grafana.enable = true;
-              };
-            };
+                enterTest = ''
+                  mix test
+                '';
+                processes.mqtt = {
+                  exec = "${pkgs.mosquitto}/bin/mosquitto -p ${toString mosquitto_port}";
+                };
+                process.managers.process-compose = {
+                  port = process_compose_port;
+                  tui.enable = true;
+                };
+                services.postgres = {
+                  enable = true;
+                  package = pkgs.postgresql_16; # 17 is not yet available in nixpkgs
+                  listen_addresses = "127.0.0.1";
+                  port = postgres_port;
+                  initialDatabases = [{ name = "teslamate"; }];
+                  initialScript = ''
+                    CREATE USER teslamate with encrypted password 'your_secure_password_here';
+                    GRANT ALL PRIVILEGES ON DATABASE teslamate TO teslamate;
+                    ALTER USER teslamate WITH SUPERUSER;
+                  '';
+                };
+              }
+            ];
 
-            testScript = ''
-              server.wait_for_open_port(4000)
+          };
+
+          moduleTest =
+            (nixpkgs.lib.nixos.runTest {
+              hostPkgs = pkgs;
+              defaults.documentation.enable = false;
+              imports = [
+                {
+                  name = "teslamate";
+                  nodes.server = {
+                    nixosModules.default = import ./module.nix { inherit self'; };
```

Comment:
Looks like this line got moved to the wrong spot after the flake-parts change? This `nodes.server` section is a NixOS config so `nixosModules.default` does nothing here. (this actually doesn't even work but I'm guessing we don't see the error since nix lazy-evaluates things and we don't run `nix flake check` in CI and this test only runs on linux systems). Also `inherit self'` won't work since `module.nix` is expecting the normal flake `self`, not `self'` from `perSystem`.

`nixosModules.default` needs to be set as `flake.nixosModules.default` in the top-level attribute set passed to the `mkFlake` call. 

@brianmay this is causing the error you just mentioned were seeing.

Reply:
Thanks for reminding, stumbled over this as well and forgot about it during illness.